### ISSUE TITLE
Split ppc/riscv assemblers

### ISSUE THREAD

**Issue by AntonKozlov** [azul-research/jdk-riscv#17]

**Comment by AntonKozlov** [comment]
One task is to gradually add riscv encodings and remove ppc ones. So for the issue, we want relatively robust separation of PPC64/RISC-V assemblers. 

It would be also nice to see how much of ppc assembler is still used by code generators.

As a sub task, we want to know if any of ppc64 instruction is attempted to execute. Some of them may be valid risc-v instructions, so we can't rely it will SIGILL. One way is to make all ppc instruction encode some valid RISC-V `bad instruction`.

**Comment by Gammanik** [comment]
I'm trying to build Jdk using these instructions: https://github.com/azul-research/jdk-riscv/blob/riscv/dev-riscv/docs/zero-compiling.md

But when I run command ./dev-riscv/scripts/make.sh -v zero in toolchain docker-container I'm getting the following error:


```
/jdk-riscv/src/hotspot/os_cpu/linux_zero/orderAccess_linux_zero.hpp: Assembler messages:
/jdk-riscv/src/hotspot/os_cpu/linux_zero/orderAccess_linux_zero.hpp:81: Error: unrecognized opcode `lwsync'
make[3]: *** [lib/CompileJvm.gmk:177: /jdk-riscv/build/linux-riscv64-zero-release/hotspot/variant-zero/libjvm/objs/abstractInterpreter.o] Error 1
make[2]: *** [make/Main.gmk:249: hotspot-zero-libs] Error 2

ERROR: Build failed for target 'default (exploded-image)' in configuration 'linux-riscv64-zero-release' (exit code 2) 

=== Output from failing command(s) repeated here ===
* For target hotspot_variant-zero_libjvm_objs_abstractInterpreter.o:
/jdk-riscv/src/hotspot/os_cpu/linux_zero/orderAccess_linux_zero.hpp: Assembler messages:
/jdk-riscv/src/hotspot/os_cpu/linux_zero/orderAccess_linux_zero.hpp:81: Error: unrecognized opcode `lwsync'

* All command lines available in /jdk-riscv/build/linux-riscv64-zero-release/make-support/failure-logs.
=== End of repeated output ===

No indication of failed target found.
Hint: Try searching the build log for '] Error'.
Hint: See doc/building.html#troubleshooting for assistance.

make[1]: *** [/jdk-riscv/make/Init.gmk:312: main] Error 2
make: *** [/jdk-riscv/make/Init.gmk:186: default] Error 2
```


I've been trying both with scripts and manual. But when I'm commenting the lines with 'lwsync' in 'src/hotspot/os_cpu/linux_zero/orderAccess_linux_zero.hpp' of course it's not being compiled.

```
#ifdef __NO_LWSYNC__
#define LIGHT_MEM_BARRIER __asm __volatile ("sync":::"memory")
#else
#define LIGHT_MEM_BARRIER __asm __volatile ("lwsync":::"memory")
#endif

#else // PPC
```

**Comment by AntonKozlov** [comment]
> But when I run command ./dev-riscv/scripts/make.sh -v zero in toolchain docker-container I'm getting the following error:

Please try use default variant (`core`). `zero` is totally assembler-less, you'll get no asm to work on

**Comment by Gammanik** [comment]
ok. Thank you

**Comment by AntonKozlov** [comment]
an ambiguity between PPC/RV assemblers creates such simple but annoying bugs: https://github.com/azul-research/jdk-riscv/pull/31/commits/00200ebe344e54486011b446d85596f86a2d5ad9

**Comment by AntonKozlov** [comment]
I think was fixed by #34 and #35. Thank you!
